src/services/member/plugins/data/controller.ts

```diff
@@ -0,0 +1,42 @@
+import { StatusCodes } from 'http-status-codes';
+
+import { FastifyPluginAsync } from 'fastify';
+
+import { buildRepositories } from '../../../../utils/repositories';
+import { DataMemberService } from './service';
+
+const plugin: FastifyPluginAsync = async (fastify) => {
+  const {
+    files: { service: fileService },
+    mailer,
+    db,
+  } = fastify;
+  const dataMemberService = new DataMemberService();
+
+  // download all related data to the given user
+  fastify.post(
+    '/',
+    {
+      preHandler: fastify.verifyAuthentication,
+    },
+    async ({ member }, reply) => {
+      db.transaction(async (manager) => {
+        const repositories = buildRepositories(manager);
+
+        await dataMemberService.requestDataExport({
+          actor: member,
+          repositories,
+          fileService,
+          mailer,
+        });
+
+        // TODO: no need to web sockets ?
```

Comment:
Hum no, I think best is a notifier in the frontend to say that the user needs to wait some time before receiving the mail.

Reply:
Nice, I was thinking a same mechanism ! Perfect !